pasqal_cloud/__init__.py

```diff
@@ -161,6 +163,18 @@ def create_batch(
             BatchCreationError: If batch creation failed
             BatchFetchingError: If batch fetching failed
         """
+        if complete is not None and open is not None:
+            raise OnlyCompleteOrOpenCanBeSet
```

Comment:
I understand the want for the exception but could we simply not ignore `complete` if `open` is set? while still outputting the warning since it is a deprecated field

Reply:
The ticket specified the following:
```
n the SDK we should accept both a complete and open kwarg. If the user sets both, raise an error. Convert these kwargs into the "open" attribute for the batch creation request.
```

Reply:
As the current behaviour in the POST batch endpoint is to raise an exception as per the specifications of the tickets and it has already been released to prod I would rather keep this one consistent and raise the exception rather than ignoring complete.

Reply:
The ticket specs can be challenged if we find a better solution! I like the suggestion as well, it makes the interface a bit more lenient on the user inputs

Reply:
It would require an MR on public-apis side to make for a consistent behaviour which I don't think this is worth but if there's a consensus I'll abide

Reply:
As a last check that this is what we want. Because I believe we're trading leniency for clarity:
The args on the batch would look like:

```
...
open: bool = False
complete: bool = True
```

Then the following conditions would lead to the following states.

- open: True , complete: True -> the batch is open, complete is ignored
- open: False (default), complete: True (default) -> The batch is closed
- open: True , complete: False - > the batch is closed, complete is ignored
- open: False (default), complete: False -> The batch is opened, (here open is ignored)


@MohamedAminMallek @oliver-gordon @MatthieuMoreau0 let me know if I got this right

Reply:
Thanks for pointing this out. 

On second thought I indeed prefer raising an exception. For old scripts running on the latest SDK version there will be no error since it will only specify `complete` - this is good for backwards compatibility's sake. If the user makes an effort to use the new `open` argument we can expect them to get rid of the old `complete` one.

That should also simplify the logic

Reply:
I am in favor of raising an exception as well.